mathesar/signals.py

```diff
@@ -27,3 +28,11 @@ def create_table_settings(**kwargs):
 def compute_preview_column_settings(**kwargs):
     instance = kwargs['instance']
     _compute_preview_template(instance.table)
+
+
+@receiver(post_save, sender=DatabaseRole)
+def give_manager_access_to_public_schema(**kwargs):
+    if kwargs['created']:
+        instance = kwargs['instance']
+        public_schema = Schema.objects.get(oid=2200, database=instance.database)
+        SchemaRole.objects.create(schema=public_schema, user=instance.user, role=Role.MANAGER.value)
```

Comment:
I'm not sure Manager permissions are appropriate given changes in Postgres. https://www.enterprisedb.com/blog/new-public-schema-permissions-postgresql-15. I'd stick to Viewer. @mathemancer might have an opinion here, in which case I'll defer to him.

Reply:
I completely removed it, it does not make sense with our permission rules. According to our permissions system, you would already have `Viewer` access to all the schemas if you have `Viewer` permission for the Database.